bindings/gcp/bucket/metadata.yaml

```diff
@@ -0,0 +1,106 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: gcp.bucket
+version: v1
+status: alpha
+title: "GCP Storage Bucket"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/gcpbucket/
+binding:
+  output: true
+  operations:
+    - name: create
+      description: "Create an item."
+capabilities: []
+authenticationProfiles:
+  - title: "GCP API call Authentication"
+    description: |
+      Authenticate authenticates API calls with the given service account or refresh token JSON credentials.
+    metadata:
+      - name: private_key_id
+        required: true
+        sensitive: true
+        description: |
+          GCP private key id.
+        example: '"privateKeyId"'
+      - name: private_key
+        required: true
+        sensitive: true
+        description: |
+          GCP credentials private key. Replace with x509 cert.
+        example: '"12345-12345"'
+metadata:
+  - name: bucket
+    required: true
+    description: |
+      The bucket name.
+    example: '"mybucket"'
+    type: string
+  - name: type
+    type: string
+    required: true
+    description: |
+      The GCP credentials type.
+    example: '"service_account"'
+  - name: project_id
+    type: string
+    required: true
+    description: |
+      GCP project id.
+    example: '"projectId"'
+  - name: client_email
+    type: string
+    required: true
+    description: |
+      GCP client email.
+    example: '"client@email.com"'
+  - name: client_id
+    type: string
+    required: true
+    description: |
+      GCP client id.
+    example: '"0123456789-0123456789"'
+  - name: auth_uri
+    type: string
+    required: false
+    description: |
+      Google account OAuth endpoint.
+    example: '"https://accounts.google.com/o/oauth2/auth"'
+  - name: token_uri
+    type: string
+    required: false
+    description: |
+      Google account token uri.
+    example: '"https://oauth2.googleapis.com/token"'
+  - name: auth_provider_x509_cert_url
+    type: string
+    required: false
+    description: |
+      GCP credentials cert url.
+    example: '"https://www.googleapis.com/oauth2/v1/certs"'
+  - name: client_x509_cert_url
```

Comment:
Why do some options use camelCase and some use snake_case? Can we make sure we are consistent? In Dapr we normally use camelCase (which our parsers handle in a case-insensitive way)

Reply:
@ItalyPaleAle For the Authentication specific fields/properties, it looks like they follow the snake_case naming convention as it appears in GCP service account keys. Here is an example of a Service Account Key JSON: 
```
{
  "type": "service_account",
  "project_id": "someproject-123456",
  "private_key_id": "4d5ac58cc2b5a01e23223123341284",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEv5vu8sUw+2/y/MfhhkIKQTgnoiIZFHximeQN67VF9pifNR/D0k\nA9W9wIr1sJJ4LITvgiui0GqzsbvW98KqeuXll1qJVlVHVSp64gMEYVSVZAK47vNs\nVP2HbvhpWdhBcKRFHs06BPyD\n-----END PRIVATE KEY-----\n",
  "client_email": "someone@someproject-123456.iam.gserviceaccount.com",
  "client_id": "1234567890",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/someproject-123456.iam.gserviceaccount.com"
}
```
I looked at all the GCP components Dapr has at the moment and they all follow the same naming pattern.